src/aws/bin/terraform

```diff
@@ -2,12 +2,54 @@
 
 set -eo pipefail
 
-# Run Terraform commands in the docker container passing our environment variables
-
-docker run --rm -it \
-    -u "$(id -u)" \
-    -v "${PWD}/:/app" \
-    -w "/app" \
-    --env-file ./env.d/development \
-    "hashicorp/terraform:0.14.5" \
-    "$@"
+declare MARSHA_TERRAFORM_ENV_FILE
+
+function terraform() {
+    current_workspace=$(
+        docker run --rm -i \
+            -u "$(id -u)" \
+            -v "${PWD}/:/app" \
+            -w "/app" \
+            "hashicorp/terraform:0.14.5" \
+            workspace show
+    ) 
+
+    # If an environment file is specified, it must match the current workspace
+    if [[ -n ${MARSHA_TERRAFORM_ENV_FILE+x} ]]; then
```

Comment:
I'm not familiar with the "+x" and didn't get what case are you trying to catch?

Reply:
I found it here https://stackoverflow.com/a/13864829
I test if the variable is not empty

Reply:
My understanding was that the difference with `if [[ -n ${MARSHA_TERRAFORM_ENV_FILE+x} ]];` is that if it is set to null, we will still consider it to be set. But that's not really what we want here?

Reply:
Yes exact, in a first implementation I did I just wanted to test if the variable is set. This is not the case anymore :+1: